tests/test_callprofiler.py

```diff
@@ -284,3 +284,91 @@ def baz():
+
+        # There are 3 keys in the serialization_dict
+        self.assertEqual(serialization_dict["id_map"], {})
+        self.assertEqual(serialization_dict["unique_id"], 0)
+        self.assertIn("radix_tree", serialization_dict)
+
+        radix_tree = serialization_dict["radix_tree"]
+        self.assertEqual(radix_tree["key"], -1)
+        self.assertEqual(radix_tree["name"], "")
+        self.assertEqual(radix_tree["call_count"], 0)
+        self.assertEqual(radix_tree["total_time"], 0)
+        self.assertEqual(radix_tree["children"], [])
+
+    def test_two_callers_serializing(self):
+
+        @profile_function
+        def bar():
+            busy_loop(0.5)
+
+        @profile_function
+        def foo():
+            busy_loop(0.1)
+            bar()
+
+        modmesh.call_profiler.reset()
+        bar()
+        foo()
+        serialization_string = modmesh.call_profiler.serialize()
+        serialization_dict = json.loads(serialization_string)
+        print(serialization_string)
+
+        # There are 3 keys in the serialization_dict
+        self.assertEqual(serialization_dict["id_map"], {'bar': 0, 'foo': 1})
+        self.assertEqual(serialization_dict["unique_id"], 3)
+        self.assertIn("radix_tree", serialization_dict)
+
+        radix_tree = serialization_dict["radix_tree"]
+        self.assertEqual(radix_tree["key"], -1)
+        self.assertEqual(radix_tree["name"], "")
+        self.assertEqual(radix_tree["call_count"], 0)
+        self.assertEqual(radix_tree["total_time"], 0)
+        self.assertIn("children", radix_tree)
+
+        children = radix_tree["children"]
+        self.assertEqual(len(children), 2)
+
+        bar_child = children[0]
+        self.assertEqual(bar_child["key"], 0)
+        self.assertEqual(bar_child["name"], "bar")
+        self.assertEqual(bar_child["call_count"], 1)
+        self.assertGreaterEqual(bar_child["total_time"], 5e8)
+        self.assertEqual(bar_child["children"], [])
+
+        foo_child = children[1]
+        self.assertEqual(foo_child["key"], 1)
+        self.assertEqual(foo_child["name"], "foo")
+        self.assertEqual(foo_child["call_count"], 1)
+        self.assertGreaterEqual(foo_child["total_time"], 1e8)
+        self.assertEqual(len(foo_child["children"]), 1)
+
+        foo_bar_child = foo_child["children"][0]
+        self.assertEqual(foo_bar_child["key"], 0)
+        self.assertEqual(foo_bar_child["name"], "bar")
+        self.assertEqual(foo_bar_child["call_count"], 1)
+        self.assertGreaterEqual(foo_bar_child["total_time"], 5e8)
+        self.assertEqual(foo_bar_child["children"], [])
+
+    def test_serialize_during_profiling(self):
+
+        @profile_function
+        def bar():
+            busy_loop(0.5)
+
+        @profile_function
+        def foo():
+            busy_loop(0.1)
+            modmesh.call_profiler.serialize()
+            bar()
+
+        modmesh.call_profiler.reset()
+        self.assertRaises(RuntimeError, foo)
```

Comment:
Here is the unit test of the function `serializtion`.

Reply:
Pretty good

Reply:
Please use [`assertRaisesRegex()`](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaisesRegex) to test for the exception message as well.